assets/system/game_modules/kami.mjs

```diff
@@ -30,34 +30,87 @@
  *  POSSIBILITY OF SUCH DAMAGE.
 **/
 
-import from from 'from';
-
 export default new
 class Kami
 {
-	constructor(title = Sphere.Game.name)
+	constructor()
 	{
 		this.enabled = SSj.now() > 0;
-		this.title = title;
 		this.records = [];
+		this.outputBordered = false;
+		this.placeholder = new Record();
 		if (this.enabled)
 			this.exitJob = Dispatch.onExit(() => this.finish());
 	}
 
 	finish()
 	{
-		SSj.log(`Profiling has completed for "${this.title}"`);
-		let sortedRecords = from(this.records)
-			.where(it => it.count > 0)
-			.descending(it => it.totalTime / it.count);
-		for (const record of sortedRecords) {
-			let averageTime = Math.round(record.totalTime / record.count / 1000).toLocaleString();
-			let count = record.count.toLocaleString();
-			let totalTime = Math.round(record.totalTime / 1000).toLocaleString();
-			SSj.log(`  ${count} occurrences "${record.description}" took ${totalTime} mcs (avg. ${averageTime} mcs)`);
-			record.target[record.methodName] = record.originalFunction;
-		}
+		if (!this.enabled)
+			return;
+		
 		this.exitJob.cancel();
+
+		let totalTime = 0;
+		let totalAverage = 0;
+		for (let i = 0; i < this.records.length; ++i) {
+			if (this.records[i].count === 0) {
+				this.records.splice(i, 1);
+				--i;
+			}
+			else {
+				const record = this.records[i];
+				record.averageTime = record.totalTime / record.count;
+				totalTime += record.totalTime;
+				totalAverage += record.averageTime;
+			}
+		}
+
+		this.records.sort(function(a, b) { return b.averageTime - a.averageTime; });
+
+		let consoleOutput = [
+			[ "Event" ],
```

Comment:
Hmm... so this array is column-major (i.e. `consoleOutput[column][row]`)?  I'm generally used to setting up 2D arrays the other way around, so this feels strange to me.

Reply:
For outputting a table in this form have to iterate down the columns so this is just much much easier - code would be messier/weirder if table was row major.